pc/storage/kdtree/kdtree_test.go

```diff
@@ -237,6 +536,61 @@ func TestKDtree_randomCloud(t *testing.T) {
 	}
 }
 
+func TestKDtree_findMinimum_randomCloud(t *testing.T) {
+	const (
+		nClouds = 1000
+		nPoints = 100
+		width   = 10.0
+	)
+
+	for i := 0; i < nClouds; i++ {
+		pp := generateRandomCloud(t, nPoints, width)
+		it, err := pp.Vec3Iterator()
+		if err != nil {
+			t.Fatal(err)
+		}
+		kdt := New(it)
+		ns := &naiveSearch{it}
+		for dim := 0; dim < 3; dim++ {
+			nodeIDKDTree, err := kdt.findMinimumImpl(kdt.root, dim, 0)
+			if err != nil {
+				t.Fatal(err)
+			}
+			nodeIDNaive := ns.findMinimum(dim)
+			if nodeIDKDTree != nodeIDNaive {
+				t.Fatalf("dim %d: Expected: %d, got %d", dim, nodeIDNaive, nodeIDKDTree)
+			}
+		}
+	}
+}
+
+func TestKDtree_DeletePoint_randomCloud(t *testing.T) {
```

Comment:
I meant to check validity of the tree its-self after deleting some points.

1. Create random cloud
2. Create kdtree and naive searcher
3. Randomly delete some points from kdtree and naive searcher
4. Search nearest point of random positions by kdtree and naive searcher and compare

Reply:
What about 80947dec5aa7f78bb9a528b6d11bf23c091a963e?